x-pack/plugins/watcher/server/routes/api/watch/register_visualize_route.ts

```diff
@@ -19,19 +19,24 @@ const bodySchema = schema.object({
   options: schema.object({}, { unknowns: 'allow' }),
 });
 
-function fetchVisualizeData(dataClient: ILegacyScopedClusterClient, index: any, body: any) {
-  const params = {
-    index,
-    body,
-    ignoreUnavailable: true,
-    allowNoIndices: true,
-    ignore: [404],
-  };
-
-  return dataClient.callAsCurrentUser('search', params);
+function fetchVisualizeData(dataClient: IScopedClusterClient, index: any, body: any) {
+  return dataClient.asCurrentUser
+    .search(
+      {
+        index,
+        body,
+        allow_no_indices: true,
```

Comment:
I guess the decision to omit `ignoreUnavailable` was deliberate?

Reply:
Same as above, will re-add